frontend/scripts/metrica.js

```diff
@@ -0,0 +1,269 @@
+(() => {
+    class requestService {
+        post(url, data, headers = {}) {
+            return fetch(url, {
+                method: 'POST',
+                crossDomain: true,
+                headers: headers,
+                body: data
+            });
+        }
+    }
+    const state = {
+        host: 'http://stage.metrica.fun/api/v1',
+        routes: {
+            create_visitor: '/visitors',
+            create_visit: '/visits'
+        }
+    };
+    window._metricaTracking = {
+        initialize() {
+            let token = this.getToken();
+            if (token) {
+                this.createVisit();
+            } else {
+                this.createVisitor(this.getTrackById())
+                    .finally(() => this.createVisit());
+            }
+        },
+        storage() {
+            return window[this.isStorage()];
+        },
+        isStorage() {
+            try {
+                localStorage.setItem('test-local-storage', 1);
+                localStorage.removeItem('test-local-storage');
+                return 'localStorage';
+            } catch (e) {
+                return 'sessionStorage';
+            }
+        },
+        getTrackById() {
+            let myScript = document.getElementById('metrica');
```

Comment:
metrica is not unique name, and some sites can use this id for their needs and then it will break a logic. Query selector still better, or id should be 100% unique